storage_controller/src/service.rs

```diff
@@ -1090,12 +1107,20 @@ impl Service {
             delayed_reconcile_tx,
             abort_tx,
             startup_complete: startup_complete.clone(),
+            background_operations_controller: Default::default(),
             cancel,
             gate: Gate::default(),
             tenant_op_locks: Default::default(),
             node_op_locks: Default::default(),
         });
 
+        let (controller, node_operations_receiver) =
+            background_node_operations::Controller::new(this.clone());
+
+        this.background_operations_controller
+            .set(controller)
```

Comment:
> a hint that the controller doesn't make sense as a separate type

Yeah, I went back and forth on this. I wanted to avoid extending the responsibility of `Service`, but the drain and fill algos ended up there anyway. Gonna try and move everything into service and see if that's preferable.

Reply:
Removed the `Controller` in [7b0c355](https://github.com/neondatabase/neon/pull/8014/commits/7b0c355d6de81cfa67740a517f2a1aa68bfb66f9)